### PR TITLE
workerpool: add skip register option

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: ref #42511

Problem Summary:

The resource control component is immature, it is better to support setting the worker size manually.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by wjhuang2016 — APPROVED** [review]

File: resourcemanager/pool/workerpool/workerpool.go [hunk 1]
```diff
@@ -42,10 +42,25 @@ type WorkerPool[T any] struct {
 	createWorker func() Worker[T]
```
Comment (hawkingrei) [inline resourcemanager/pool/workerpool/workerpool.go:42]: If you want a work pool which is used in special, not to the scheduler, you only skip it in the scheduler in the resource mange.

Comment (hawkingrei) [reply resourcemanager/pool/workerpool/workerpool.go:42]: ```Tune(_ util.Component, pool util.GoroutinePool)``` 

In the scheduler, you can get the type of component and return hold to skip schedule.

Comment (hawkingrei) [reply resourcemanager/pool/workerpool/workerpool.go:42]: I suggest you can write your scheduler for your task. CPU scheduler is a toy.

Comment (tangenta) [reply resourcemanager/pool/workerpool/workerpool.go:42]: OK, we can implement the automatic tuning later.

**Pull request merged**